src/main/java/seedu/address/model/library/Library.java

```diff
@@ -120,7 +129,11 @@ public ObservableList<Book> getBookList() {
     }
 
     public void setBookList(ObservableList<Book> bookList) {
-        this.bookList = bookList;
+        ObservableList<Book> expectedBookList = FXCollections.observableArrayList();
+        for (Book book : bookList) {
+            expectedBookList.add(book);
+        }
+        this.bookList = expectedBookList;
```

Comment:
When fixing the test cases, we realised that the constructor does not return a new object. As a result, when making changes to the expected model, the changes also reflect in the original model. Thus, we made this change in order to have our test cases work as expected.